lzero/policy/unizero.py

```diff
@@ -0,0 +1,946 @@
+            'analysis/last_step_loss_policy': last_step_losses['loss_policy'].item(),
+            'analysis/last_step_loss_rewards': last_step_losses['loss_rewards'].item(),
+            'analysis/last_step_loss_obs': last_step_losses['loss_obs'].item(),
+
+            'Current_GPU': current_memory_allocated_gb,
+            'Max_GPU': max_memory_allocated_gb,
+            'collect_mcts_temperature': self._collect_mcts_temperature,
+            'collect_epsilon': self.collect_epsilon,
+            'cur_lr_world_model': self._optimizer_world_model.param_groups[0]['lr'],
+            'weighted_total_loss': weighted_total_loss.item(),
+            'obs_loss': obs_loss,
+            'latent_recon_loss': latent_recon_loss,
+            'perceptual_loss': perceptual_loss,
+            'policy_loss': policy_loss,
+            'orig_policy_loss': orig_policy_loss,
+            'policy_entropy': policy_entropy,
+            'target_policy_entropy': average_target_policy_entropy,
+            'reward_loss': reward_loss,
+            'value_loss': value_loss,
+            'value_priority_orig': np.zeros(self._cfg.batch_size),  # TODO
+            'target_reward': target_reward.mean().item(),
+            'target_value': target_value.mean().item(),
+            'transformed_target_reward': transformed_target_reward.mean().item(),
+            'transformed_target_value': transformed_target_value.mean().item(),
+            'total_grad_norm_before_clip_wm': total_grad_norm_before_clip_wm.item(),
+            'analysis/dormant_ratio_encoder': dormant_ratio_encoder,
+            'analysis/dormant_ratio_world_model': dormant_ratio_world_model,
+            'analysis/latent_state_l2_norms': latent_state_l2_norms,
+            'analysis/l2_norm_before': self.l2_norm_before,
+            'analysis/l2_norm_after': self.l2_norm_after,
+            'analysis/grad_norm_before': self.grad_norm_before,
+            'analysis/grad_norm_after': self.grad_norm_after,
+        }
+
+        return return_loss_dict
+
+    def monitor_weights_and_grads(self, model):
+        for name, param in model.named_parameters():
+            if param.requires_grad:
+                print(f"Layer: {name} | "
+                      f"Weight mean: {param.data.mean():.4f} | "
+                      f"Weight std: {param.data.std():.4f} | "
+                      f"Grad mean: {param.grad.mean():.4f} | "
+                      f"Grad std: {param.grad.std():.4f}")
+
+    def _init_collect(self) -> None:
+        """
+        Overview:
+            Collect mode init method. Called by ``self.__init__``. Initialize the collect model and MCTS utils.
+        """
+        self._collect_model = self._model
+
+        if self._cfg.mcts_ctree:
+            self._mcts_collect = MCTSCtree(self._cfg)
+        else:
+            self._mcts_collect = MCTSPtree(self._cfg)
+        self._collect_mcts_temperature = 1.
+        self.collect_epsilon = 0.0
```

Comment:
rename to the variable in policy startswith `_`, such as `self._collect_epsilon`